### ISSUE TITLE
java.lang.SecurityException: uid 10204 cannot get user data for accounts of type: nextcloud

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#1248]
Received via mail and also reported here https://help.nextcloud.com/t/call-for-testing-android-client-version-1-5/18755/7

```
07-28 13:58:12.172 13198 13507 E RemoteOperation: Error while trying to access to SERVER/account
07-28 13:58:12.172 13198 13507 E RemoteOperation: java.lang.SecurityException: uid 10204 cannot get user data for accounts of type: nextcloud
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at android.os.Parcel.readException(Parcel.java:1599)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at android.os.Parcel.readException(Parcel.java:1552)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at android.accounts.IAccountManager$Stub$Proxy.getUserData(IAccountManager.java:719)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at android.accounts.AccountManager.getUserData(AccountManager.java:365)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at com.owncloud.android.lib.common.accounts.AccountUtils.restoreCookies(AccountUtils.java:300)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at com.owncloud.android.lib.common.accounts.AccountUtils.restoreCookies(AccountUtils.java:342)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at com.owncloud.android.lib.common.SingleSessionManager.getClientFor(SingleSessionManager.java:123)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at com.owncloud.android.lib.common.DynamicSessionManager.getClientFor(DynamicSessionManager.java:40)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:113)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at com.owncloud.android.operations.common.SyncOperation.execute(SyncOperation.java:72)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at com.owncloud.android.authentication.AccountUtils$1.run(AccountUtils.java:178)
07-28 13:58:12.172 13198 13507 E RemoteOperation: 	at java.lang.Thread.run(Thread.java:818)
```

I installed on an emulator 1.4.3 and latest dev version in parallel and played a bit around, but could not trigger this error.

**Comment by tobiasKaminsky** [comment]
Ideally this should be done for next version.
But I need more infos about the setting if this is happening as I cannot reproduce it myself.

**Comment by mario** [comment]
Maybe different APK signatures like I mentioned in another issue? Have no idea otherwise if we have access to account manager...

**Comment by tobiasKaminsky** [comment]
Maybe. Do you know if this leads to a crash? Or "only" some output in the logcat?

**Comment by mario** [comment]
@tobiasKaminsky newest APK overwrites the dev version since it's the same package name (.beta), but yes, you do need to uninstall due to signature issues.... 

Maybe the dev version creates the account, and regular one tries to access it? (or vice versa?)

**Comment by mario** [comment]
Or maybe we should see if we check if we have access to account manager everywhere?

**Comment by tobiasKaminsky** [comment]
>  String cookiesString = am.getUserData(account, "oc_account_cookies"); 

this seems to be the line which throws the error.
https://developer.android.com/reference/android/accounts/AccountManager.html#getUserData(android.accounts.Account, java.lang.String) says:
`This method requires the caller to have a signature match with the authenticator that owns the specified account.`

I installed in parallel:
- nextcloud dev
- nextcloud 1.4.2
- owncloud 2.4.0
all with the same server/user and played around a bit:
- browsing files
- downloading files

I cannot reproduce it :-/

Yes, we can add a check to it. But this would just prevent the exception (and thus cannot use the cookies) and not fix the real cause of the problem. So we need to know how to reproduce this.

**Comment by mario** [comment]
Well signature match does sound like what I mentioned? Lets just make sure
it doesnt crash and doesnt use cookies in this case.

**Comment by tobiasKaminsky** [comment]
Do we know someone who can reproduce this issue? It is very important to know with which apps this happens.
I am not sure if the app works with completely no cookies.

My bet is that the system gets confused by the different devs version we have...

**Comment by mario** [comment]
Well I'm fine with releasing beta as-is without this and see if it repeats itself...

Please review and merge all other PRs targetted at 1.5 then so we can do this today.
cc @AndyScherzinger

**Comment by AndyScherzinger** [comment]
This is the last remaining issue besides the lint fixes for a initial RC for the GPlay beta channel

**Comment by mario** [comment]
We are kindof ignoring thus one. What about search refactor?

**Comment by tobiasKaminsky** [comment]
I am not ignoring it, but 
> Do we know someone who can reproduce this issue? 
and
> I am not sure if the app works with completely no cookies.

So I have no clue currently and no idea how to properly circumvent this.

**Comment by AndyScherzinger** [comment]
So postpone it?

**Comment by tobiasKaminsky** [comment]
I think it is related to the different app version we had (dev vs auto upload dev), so I would say that we go for beta and see if it happens again.

**Comment by mario** [comment]
It would be possible to do ifs for that as well ;)

**Comment by AndyScherzinger** [comment]
Sounds good to me! 👍

**Comment by Pabubung** [comment]
Hello I have problem when trying to remove google account from my android device

java.lang.SecurityException: uid 10132 cannot remove accounts of type: com.google

How can I fix this problem. Thanks
